### PR TITLE
Add paho mqtt

### PR DESCRIPTION
**Pull request by blavka**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -19,11 +22,14 @@ int main(int argc, char **argv)
...
-            { "furious", 'f',      0,         0, "Do not wait for the initial start string" },
-            { "log",     'L',      "level",   0, "Set desired log level to one of the following options: dump|debug|info|warning|error|fatal" },
-            { "version", 'v',      0,         0, "Give version information" },
+            { "list",   OPTION_LIST,   0,         0, "Show list of available devices" },
+            { "dev",    OPTION_DEV,    "ID|PATH", 0, "Select device by ID or PATH (beware that ID may change with USB ports' manipulation)" },
+            { "host",   OPTION_HOST,   "HOST",    0, "MQTT host to connect to" },
+            { "port",   OPTION_PORT,   "PORT",    0, "MQTT port to connect to (default is 1883)" },
+            { "prefix", OPTION_PREFIX, "PREFIX",  0, "MQTT prefix topic" },
+            { "furious", 'f',          0,         0, "Do not wait for the initial start string" },
+            { "log",     'L',          "level",   0, "Set desired log level to one of the following options: dump|debug|info|warning|error|fatal" },
```
Comment (hubpav) [inline src/main.c]: Please change "level" to "LEVEL" to make it orthogonal to HOST, PORT, etc.

File: src/bc_talk.c [hunk 2]
```diff
@@ -37,45 +53,70 @@ static void *bc_talk_worker_stdin(void *parameter);
...
 
-    bc_talk_callback = callback;
+void bc_talk_init_mqtt(bc_talk_parse_callback callback, char *host, int port, char *prefix)
+{
+    _bc_talk_init(callback);
 
-    memset(bc_talk_device_names, 0, sizeof(bc_talk_device_names));
+    bc_talk_mqtt = true;
+    bc_talk_mqtt_prefix = prefix;
+    strcpy(bc_talk_msq.topic, bc_talk_mqtt_prefix);
```
Comment (hubpav) [inline src/bc_talk.c]: I have not examined this in detail but in general try to stick with strncpy, strncat, snprintf, atc. string function variants for safety reasons... You may prevent buffer overflow problems in many situations :smile: Put it to your TODO list and refactor the whole code base to use these safe variants across the whole code base.

File: src/bc_talk.c [hunk 3]
```diff
@@ -37,45 +53,70 @@ static void *bc_talk_worker_stdin(void *parameter);
...
+    sprintf(serverURI, "tcp://%s:%d", host, port);
 
-    bc_os_mutex_init(&bc_talk_mutex);
-    bc_os_task_init(&bc_talk_task_stdin, bc_talk_worker_stdin, NULL);
+    MQTTAsync_create(&bc_talk_mqtt_client, serverURI, "clientID", MQTTCLIENT_PERSISTENCE_NONE, NULL);
+
+    MQTTAsync_setCallbacks(bc_talk_mqtt_client, NULL, _bc_talk_mqtt_connection_lost_callback,
+                           _bc_talk_mqtt_message_callback, NULL);
+
+    MQTTAsync_connectOptions conn_opts = MQTTAsync_connectOptions_initializer;
```
Comment (hubpav) [inline src/bc_talk.c]: BTW: This is an inspiring construct with initializer. I may consider this idea to firmware repositories for similar situation...

File: src/bc_talk.c [hunk 4]
```diff
@@ -100,20 +141,19 @@ void bc_talk_publish_add_quantity(char *name, char *unit, char *value, ...)
...
+        bc_talk_msq.length += 2;
     }
 
-    fprintf(stdout, "\"%s\": [", name);
-
+    bc_talk_msq.length += sprintf(bc_talk_msq.payload + bc_talk_msq.length, "\"%s\": [", name);
 
     va_start(ap, value);
-    vfprintf(stdout, value, ap);
+    bc_talk_msq.length += vsprintf(bc_talk_msq.payload + bc_talk_msq.length, value, ap);
```
Comment (hubpav) [inline src/bc_talk.c]: vsnprintf...

File: CHANGELOG.adoc [hunk 5]
```diff
@@ -4,9 +4,13 @@
 == link:{gh-url}/tree/v1.1.0[1.1.0] (unreleased)
 Added::
 * Support for Humidity TAG R2.1
```
Comment (hubpav) [inline CHANGELOG.adoc]: This is old change but next time please stick with original names in human-readable texts - the name is Humidity Tag.

File: CHANGELOG.adoc [hunk 6]
```diff
@@ -4,9 +4,13 @@
...
+* Support mqtt
```
Comment (hubpav) [inline CHANGELOG.adoc]: It is MQTT :smile:

File: src/main.c (hunk 1 above)
Comment (blavka) [inline src/main.c]: Ok

File: src/bc_talk.c (hunk 2 above)
Comment (blavka) [inline src/bc_talk.c]: ok

File: src/bc_talk.c (hunk 4 above)
Comment (blavka) [inline src/bc_talk.c]: ok

File: CHANGELOG.adoc (hunk 5 above)
Comment (blavka) [inline CHANGELOG.adoc]: ok

File: CHANGELOG.adoc (hunk 6 above)
Comment (blavka) [inline CHANGELOG.adoc]: ok

**Pull request merged**